### ISSUE TITLE
Inherited settings aren't handled properly

### ISSUE THREAD

**Issue by agc93** [spectresystems/spectre.commandline#10]
Using a setup like [this sample](https://gist.github.com/agc93/a84c9daff32a50c01f135816d48141d7), I would expect that settings would be handled as follows:

- Arguments and options on the proxy command setting are applied "to" the proxy command :white_check_mark:
- Arguments and options on the child command settings class are applied to the child command :white_check_mark: 
- Arguments and options inherited "between" the proxy command class and the child command class should also be parsed and included in help :question: 

In the above example, that would mean that the Name argument would be included in the help and parsing for *both* `MyCommand`/`child` _and_ `MyOtherCommand`/`other`

Let me know if that's not clear.

**Comment by agc93** [comment]
I realised this was a confusing way of explaining things, so I created a simple demo project [here](https://github.com/agc93/spectre-cli-sample) to show what I meant. Hopefully that's a little easier to understand :smile:

**Comment by patriksvensson** [comment]
@agc93 Perfect! This is on my list to fix so an example is perfect! 😊

**Comment by patriksvensson** [comment]
@agc93 So you mean that the following should work?

```
program.exe proxy child Patrik
program.exe proxy other Patrik
```

as well as:

```
program.exe proxy Patrik other
```

Or have I misunderstood something?

**Comment by patriksvensson** [comment]
Should be resolved in version 0.6.0.
